packages/dx-grid-core/src/plugins/virtual-table-state/helpers.ts

```diff
@@ -101,8 +101,12 @@ export const trimRowsToInterval: PureComputed<[VirtualRows, Interval]> = (
 
 export const getAvailableRowCount: PureComputed<[boolean, number, number, number], number> = (
   infiniteScroll, newCount, lastCount, totalRowCount,
-) => (
-  infiniteScroll
-    ? Math.max(newCount, lastCount)
+) => {
+  const maxWhenInfiniteScrolling = Math.min(
+    Math.max(newCount, lastCount),
+    totalRowCount);
+  return (infiniteScroll
```

Comment:
If `infiniteScroll` is a boolean variable, I suggest using the next variable name - `isInfiniteScrolling`

Reply:
You have 3 variables - `newCount`, `lastCount`, `totalRowCount`. You should add the `Row` word into each name or remove this word from `totalRowCount`

Reply:
You calculate `maxWhenInfiniteScrolling` variable always but use it only `infiniteScroll` is `true`

Reply:
We should add tests for new functionality

Reply:
> You calculate `maxWhenInfiniteScrolling` variable always but use it only `infiniteScroll` is `true`

I can remove `maxWhenInfiniteScrolling` variable and calculate this in expression. But it may do code less readable. Should we do that?

Reply:
It will be more readable with implicit return